src/qofonoobject.cpp

```diff
@@ -240,6 +240,9 @@ void QOfonoObject::updateProperty(const QString &key, const QVariant &value)
             d_ptr->properties.remove(key);
         }
         propertyChanged(key, newValue);
+    } else if (!oldValue.isValid() && newValue.isValid()) {
```

Comment:
And I find it strange that `qofono::safeVariantEq` declares valid and invalid QVariant's equal each other. Is that intentional? If it is, it definitely deserves a comment.